src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/security/AuthorisedCaseSearchOperation.java

```diff
@@ -83,35 +95,75 @@ private CaseSearchResult searchCasesAndFilterFieldsByAccess(List<CaseType> autho
         }
 
         CaseSearchResult result = caseSearchOperation.execute(authorisedSearchRequest);
-        filterFieldsByAccess(authorisedCaseTypes, result.getCases());
+        filterFieldsByAccess(authorisedCaseTypes, result.getCases(), authorisedSearchRequest);
 
         return result;
     }
 
-    private void filterFieldsByAccess(List<CaseType> authorisedCaseTypes, List<CaseDetails> cases) {
+    private void filterFieldsByAccess(List<CaseType> authorisedCaseTypes, List<CaseDetails> cases, CrossCaseTypeSearchRequest authorisedSearchRequest) {
         Map<String, CaseType> caseTypeIdByCaseType = authorisedCaseTypes
             .stream()
             .collect(Collectors.toMap(CaseType::getId, Function.identity()));
 
         cases.stream()
             .filter(caseDetails -> caseTypeIdByCaseType.containsKey(caseDetails.getCaseTypeId()))
```

Comment:
@gbenadikar trying to understand why we have this filtering here. Why is that?

Reply:
simplified a bit